src/modules/fundingManager/extensions/FM_EXT_TokenVault_v1.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.23;
+
+// Internal
+import {Module_v1} from "src/modules/base/Module_v1.sol";
+import {IFM_EXT_TokenVault_v1} from
+    "src/modules/fundingManager/extensions/interfaces/IFM_EXT_TokenVault_v1.sol";
+
+// External
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {ERC165Upgradeable} from
+    "@oz-up/utils/introspection/ERC165Upgradeable.sol";
+
+/**
+ * @title   Inverter Token Vault
```

Comment:
Yes and no:
It definitely has properties of a logicmodule as you suggested, but the point here is to **not** use the paymentProcessor to transfer tokens, which I would say is one of the main points of a logicmodule.
In regards of it being a extension: For me the VotingRoles argument doesnt work here. Yes the Module extends the functionality of another module, but it also makes sense on its own. It can act as a Controller of other contracts for example and doesnt necessarily have to be linked into our authorizer setup. Broken down its just a voting mechanism contract that can call stuff. The same goes for the Vault. It may work on its own but in this context it extends the functionality of the workflow by providing a Auhtorized setup for storing tokens, that are not included in the classic token flow of the workflow
I would agree, that FM in the naming might be misleading here, but we currently dont have a system to categorize this appropriately yet.